Monika-After-Story/MonikaModDev
# Player Birthday Event

The long overdue player birthday event. This also includes a new event where you tell or confirm your birthday to Monika, replacing the old year older mood.

# Created 

- `mas_isplayer_bday()` method which returns true if it is the player's birthday

- `mas_player_bday_curr() ` which returns the player's birthday in a given year, defaults to current year

# Testing

- make sure to keep a copy of `persistent` from before you start testing so you can test multiple times if need be.

- open `options.rpy` and edit line 26 from `define config.version = "0.8.14"` to `define config.version = "0.8.15"` and save. You'll need to do this any time you pull and `options.rpy` is overwritten.

### Normal+ flow

- load in on established persistent and confirm player birthday event runs and set your birthday

- can also try with a fresh persistent and verify after you click the calendar for the first time, soon after an event runs to confirm your birthday if you have't already done so thru `monika_birthday` topic or via `mas_birthdate`

- if you set your birthday to today and she already knew your birthday was today, confirm you get birthday party with banners, balloons, cake, and birthday card, and that `Let's go out for my birthday` farewell option is present.

- if you set your birthday to today and she didn't know it was today previously, confirm no party, no banner or cake, she sings you happy birthday, and `Let's go out for my birthday` farewell option is present.

- if your birthday is not today, go to your birthday (if time jumping, suggest setting your birthday to tomorrow) and load in.  Confirm special black screen greeting with different paths depending on your choices.  All lead to full party with banners, balloons, cake, and card.  Verify `Let's go out for my birthday` farewell option is present.

- Verify decorations are still present on subsequent loads.

- Verify `Let's go on a date for my birthday` farewell option leads to special dialogue and a special greeting, along with special affection point gains, soft-capped at 20.

- Verify returning home after your birthday, you still get points and that she takes down the decorations after returning home.

- Verify decorations are not present after your birthday.

### Upset and Distressed
- Verify short happy birthday dialogue, no decorations, no special farewell. Similar event whether you confirm your birthday on your birthday or not.

### Broken

- Verify no mention, decorations or special farewell option.

### Birthday is on Halloween or Christmas

- Verify no special greeting, but instead special queued event with the same rules as above

### No restart on birthday

- Verify if you start a session before your birthday and never quit, that sometime soon after 7pm you get a special queued event that starts the party.  Same rules as above.

### Piano

- Confirm playing Happy Birthday on the piano on your birthday results in special dialogue #3454 

### History

- Verify the following are reset on January 6th so the event works correctly the following year
           `persistent._mas_player_bday_in_player_bday_mode`
           `persistent._mas_player_bday_opened_door`
           `persistent._mas_player_bday_no_decor`
           `persistent._mas_player_bday_date`
           `persistent._mas_player_bday_date_aff_gain` 
           `persistent._mas_player_bday_when_confirmed`
           `persistent._mas_player_bday_low_aff`
